lib/Fmt_ast.ml

```diff
@@ -2156,14 +2156,16 @@ and fmt_expression c ?(box = true) ?(pro = noop) ?eol ?parens
           c.conf
           (fmt_constant c const $ fmt_atrs)
   | Pexp_constraint (e, t) ->
+      let outer_parens = has_attr && parens in
```

Comment:
`has_attr` doesn't imply parentheses, as seen for example in the `override.ml.ref` test.
Whether to output parentheses is specified in `Ast.parenze_exp` (which is how `parens` is computed) but in this case, it returns true too often.
I improved the Ast rules in this branch (2 commits): https://github.com/Julow/ocamlformat/tree/2513-tweak-ast Could you merge it ?